lib/portlayer/exec/exec.go

```diff
@@ -171,6 +182,147 @@ func eventCallback(ie events.Event) {
 	return
 }
 
+func waitForContainerStarted(container *Container) bool {
+	ctx, cancel := context.WithTimeout(context.Background(), propertyCollectorTimeout)
+	defer cancel()
+
+	if container.vm == nil {
+		log.Errorf("Container(%s) wait failed for vm is not found", container.ExecConfig.ID)
+		return false
+	}
+	log.Debugf("Waiting for container(%s) started", container.ExecConfig.ID)
+	for _, k := range extraconfig.CalculateKeys(container.ExecConfig, "Sessions.*.Detail.StartTime", "") {
+		var poweredoff bool
+		var detail string
+		waitFunc := func(pc []types.PropertyChange) bool {
```

Comment:
This method should merge with vm.WaitForKeyInExtraConfig, with customized condition check as input method. But this will impact the existing function.
As right now, this method is not used by anyone else, would you mind I push this refactor out of 0.8? Then I could focus on make this PR correct today.